lib/complex.ex

```diff
@@ -6,47 +6,56 @@ defmodule Complex do
   Each complex number is represented as a structure holding the real and
   imaginary part.  There are functions for creation and manipulation of
   them.  Unfortunately since there is no operator overloading in Elixir the
-  math functions (add, subtract, etc.) are implemented as add/2, sub/2, etc.
+  math functions (add, subtract, etc.) are implemented as add/2, subtract/2, etc.
 
   ## Examples
       iex> Complex.new(3, 4)
-      %Complex{im: 4, re: 3}
+      %Complex{im: 4.0, re: 3.0}
 
       iex> Complex.imag()
       %Complex{im: 1.0, re: 0.0}
   """
 
   @vsn 2
 
-  import Kernel, except: [abs: 1, div: 2]
+  import Kernel, except: [abs: 1]
+  import Complex.CompileTimeChecks
 
   @typedoc """
   General type for complex numbers
   """
   @type complex :: %Complex{re: number, im: number}
   defstruct re: 0, im: 0
 
-  # Distinguishes between a Complex type and a number type.  This function
-  # is used to allow mixed Complex and numbers in function calls.
-  defp decompose(number) when is_number(number), do: {number, 0}
-  defp decompose(%Complex{re: re, im: im}), do: {re, im}
+  defimpl Inspect do
+    def inspect(val, opts),
+      do: Complex.to_string(val, opts.custom_options[:imaginary_constant] || "i")
+  end
+
+  def to_string(%Complex{re: re, im: im}, imaginary_constant \\ "i") do
```

Comment:
Docs missing?

I would also make the second argument an option. Specially because we may want to customize other stuff, such as removing the spaces around the operator.

Reply:
I'll resolve all doc-related comments in favor of #8

Reply:
I'll remove this extra argument for now. I'll live with `i` instead of `j` as much as it pains me hahaha